tests/test_discrete_allocation.py

```diff
@@ -230,28 +230,7 @@ def test_lp_portfolio_allocation():
     da = DiscreteAllocation(w, latest_prices, short_ratio=0.3)
     allocation, leftover = da.lp_portfolio()
 
-    #  Weirdly, this gives different answers for py3.8+ vs py3.6-3.7.
```

Comment:
I have tested with 3.7 and 3.8 and when installed through poetry, the results were the same (but different from both of the ones listed here :D ). In general, keeping tests stable across python versions seems easier to me when using poetry.
It is possible that all solutions yield a very similar objective value, so testing for that could also be an option.

Reply:
This was my mistake – both options are actually the same. I've pushed a fix (removing AMD from one of them)